### ISSUE TITLE
Auth code is empty in OnSignInSilently

### ISSUE THREAD

**Issue by gaurangbackrock** [Thaina/google-signin-unity#13]
<img width="876" alt="Screenshot 2024-07-30 at 6 52 55 PM" src="https://github.com/user-attachments/assets/e6c6002b-b1a6-4580-9ece-cbe4c1956ac9">
I use the script from the example and call OnSignInSilently(), but every time Auth code is empty.
Even ServerClientID is present.

**Comment by Thaina** [comment]
Have you set `RequestAuthCode` to true?

**Comment by gaurangbackrock** [comment]
Yes, I am using below configuration for OnSignInSilently().
configuration = new()
        {
            WebClientId = webClientId,
            RequestEmail = true,
            RequestIdToken = true,
            RequestAuthCode = true,
            RequestProfile = true,
            UseGameSignIn = false,
            HidePopups = true
        };

**Comment by Thaina** [comment]
Then could you please try to use idToken for what you are trying to do instead of authcode and see the result ?

**Comment by gaurangbackrock** [comment]
Sorry it cannot be used, I am using Playfab Integration and it needs authcode to login to the game.
I tried earlier with idToken and it gives error as it does not accept the idToken.

**Comment by Thaina** [comment]
Then could you try signin normally (not silently) to see if it can provide serverAuthCode?

**Comment by gaurangbackrock** [comment]
Yes it gives authcode, but we need to use it for first time when user sign up into the game.
Then 2nd time i need to run silently to get authcode which does not ask to user for account selection and sign in backend and then will login to playfab through authcode.

If i used directly sign in then everytime will ask account selection process when game opens which is not good and acceptable.

**Comment by Thaina** [comment]
@gaurangbackrock There was some mention about security reason that seem related https://stackoverflow.com/a/52424847/1260333

There was also suggestion to use refreshtoken API instead of request the new authcode. I think playfab has one

Is this correct? https://learn.microsoft.com/en-us/gaming/playfab/sdks/c/relogin

**Comment by gaurangbackrock** [comment]
Okay.
I understood about the authcode is not sent for 2nd time due to security reason.
But the link mention of playfab is not for Unity, it is for Azure.
I will search for Playfab login with google token instead of authcode.
Please let me know if you found something related to Playfab for Unity using google token.
Thanks for your time and suggestions.
Please also help me to solve this with idtoken, that will be great.

**Comment by Thaina** [comment]
Maybe this? google jwt seem to be based on openid 

I can't be much help because I never use playfab. Mostly use just firebase

https://learn.microsoft.com/en-us/rest/api/playfab/client/authentication/login-with-openid-connect?view=playfab-rest

**Comment by gaurangbackrock** [comment]
No, it cannot be used.
It is not related to Google Login flow with playfab.
I will search for it and if found something then will comment here.
Thanks for the suggestions.

**Comment by Thaina** [comment]
Yes, it's not relate to google login flow of playfab. But google oauth also a standard authentication and they say they conform to openid https://developers.google.com/identity/openid-connect/openid-connect

And playfab seem to have option to use openid so it has high chance to accept google's token as openid flow too

**Comment by gaurangbackrock** [comment]
I think it depend to validate on server side and use the profile to create the playfab user on server side only.
We are not using any server side api.
But will check this in detail if it helps in unity side only.
